tests/run-make/issue-107205/main.rs

```diff
@@ -0,0 +1,16 @@
+use std::fmt;
+
+pub struct Wrapper(fn(val: &()));
+
+impl fmt::Debug for Wrapper {
+    fn fmt<'a>(&'a self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("Wrapper").field(&self.0 as &fn(&'a ())).finish()
+    }
+}
+
+fn useful(_: &()) {
+}
+
+fn main() {
+    println!("{:?}", Wrapper(useful));
+}
```

Comment:
Can you try to turn this into a `build-pass` UI test? That would be more lightweight than a run-make test. Here is an example: https://github.com/rust-lang/rust/blob/9556b56dbdbd4238f0051e7230004b0aa488fa14/tests/ui/associated-types/issue-64848.rs#L1

Reply:
Moved it under `tests/ui/coercion/` directory.